### PR TITLE
Disable unix::net::ancillary on BSD.

### PR DESCRIPTION
**Pull request by m-ou-se**
See https://github.com/rust-lang/rust/issues/76915#issuecomment-1118954474

### CODE REVIEW

File: library/std/src/os/unix/net/datagram.rs [hunk 1]
```diff
@@ -1,24 +1,7 @@
-#[cfg(any(
-    doc,
-    target_os = "android",
-    target_os = "dragonfly",
-    target_os = "emscripten",
```
Comment (fbstj) [inline library/std/src/os/unix/net/datagram.rs:5]: this seems to have removed emscripten as well which isn't mentioned in the PR, was this intentional?

**Comment by klensy** [comment]
>`Merge branch 'master' into bsd-no-ancillary`

No-Merge Policy?

File: library/std/src/os/unix/net/datagram.rs (hunk 1 above)
Comment (klensy) [reply library/std/src/os/unix/net/datagram.rs:5]: > this seems to have removed emscripten as well which isn't mentioned in the PR, was this intentional?

Comment not resolved too, yes?

Comment (joshtriplett) [reply library/std/src/os/unix/net/datagram.rs:5]: I merged this intentionally knowing that it did remove other platforms as well.

**Comment by joshtriplett** [comment]
> > `Merge branch 'master' into bsd-no-ancillary`
> 
> No-Merge Policy?

Didn't realize github's conflict resolution tool would cause a problem. Should I rebase and push to the branch instead?

**Comment by klensy** [comment]
Yes, rebase + force push should work.

**Comment by joshtriplett** [comment]
@klensy @JohnTitor Done.

**Pull request merged**